### PR TITLE
feat: add controls components

### PR DESCRIPTION
**Pull request by MirandaMakharashvili**
- Radio component
- Checkbox component
- Colors component
- Counter component

### CODE REVIEW

**Comment by MirandaMakharashvili** [comment]
[Controls](https://trello.com/c/D5vZVIqc/8-controls)

File: src/app/app.component.ts [hunk 1]
```diff
@@ -1,5 +1,16 @@
 import { Component } from '@angular/core';
 import { buttonView, buttonType, iconPosition } from './core/models/button.enum';
+import { counterStyle } from 'src/app/core/models/counter.enum';
```
Comment (vsvanidze) [inline src/app/app.component.ts]: must be removed. Use git stash to use testing code comfortable

https://git-scm.com/docs/git-stash

File: src/app/components/checkbox/checkbox.component.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  @Input() list!: checkboxInterface;
+  disabled: boolean = false;
+
+  onChange = (value: any) => { };
+  onTouched = () => { };
+
+  constructor(public fb: FormBuilder) { }
+
+  ngOnInit() {
+    this.form.valueChanges.subscribe((value) => {
```
Comment (vsvanidze) [inline src/app/components/checkbox/checkbox.component.ts]: we need to unsubscribe when this component will destroy

use take until strategy   https://www.digitalocean.com/community/tutorials/angular-takeuntil-rxjs-unsubscribe

File: src/app/components/checkbox/checkbox.component.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      );
+    });
+  }
+
+  get myForm(): FormArray {
+    return this.form.controls['checkboxGroup'] as FormArray;
+  }
+
+  getControlById(): FormArray {
+    const testVal = this.form.get('checkboxGroup') as FormArray;
```
Comment (vsvanidze) [inline src/app/components/checkbox/checkbox.component.ts]: We can use shorthand syntax, return without variable

File: src/app/components/checkbox/checkbox.component.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    },
+  ]
+})
+
+export class CheckboxComponent implements ControlValueAccessor, OnInit {
+
+  requestStatus = requestStatus;
+
+  @Input() list!: checkboxInterface;
+  disabled: boolean = false;
```
Comment (vsvanidze) [inline src/app/components/checkbox/checkbox.component.ts:28]: we Must use this property

File: src/app/components/colors/colors.component.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      multi: true,
+    },
+  ]
+})
+export class ColorsComponent implements ControlValueAccessor, OnInit {
+
+  requestStatus = requestStatus;
+
+  @Input() list!: colorsInterface;
+  disabled: boolean = false;
```
Comment (vsvanidze) [inline src/app/components/colors/colors.component.ts:29]: we must use this property

File: src/app/components/counter/counter.component.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+  counterValue: number = 0;
+
+  onChange = (value: any) => { };
+  onTouche = () => { };
+
+  constructor(public fb: FormBuilder) { }
+
+  ngOnInit(): void {
+    this.counterForm.valueChanges.subscribe((value) => {
```
Comment (vsvanidze) [inline src/app/components/counter/counter.component.ts]: we need to unsubscribe in destroy

use take until strategy:  https://www.digitalocean.com/community/tutorials/angular-takeuntil-rxjs-unsubscribe

File: src/app/components/counter/counter.component.ts [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+      this.counterValue = 1;
+    }
+
+    this.onChange(this.counterValue);
+  }
+
+  plus() {
+    this.onTouche();
+
+    if (this.counterValue >= 1) {
```
Comment (vsvanidze) [inline src/app/components/counter/counter.component.ts:62]: this must be checked in form validator,

https://netbasal.com/three-ways-to-dynamically-alter-your-form-validation-in-angular-e5fd15f1e946

File: src/app/components/radio/radio.component.ts [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+      multi: true,
+    },
+  ]
+})
+export class RadioComponent implements ControlValueAccessor, OnInit {
+
+  requestStatus = requestStatus;
+
+  @Input() list!: radioInterface;
+  disabled: boolean = false;
```
Comment (vsvanidze) [inline src/app/components/radio/radio.component.ts:28]: we must use this property

File: src/app/components/radio/radio.component.ts [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+  onChange = (value: any) => { };
+  onTouched = () => { };
+
+  constructor(public fb: FormBuilder) { }
+
+  ngOnInit(): void {
+    this.radioForm.valueChanges.subscribe((value) => {
```
Comment (vsvanidze) [inline src/app/components/radio/radio.component.ts]: we need to unsubscribe when this component will destroy

use take until strategy https://www.digitalocean.com/community/tutorials/angular-takeuntil-rxjs-unsubscribe

File: src/app/components/colors/colors.component.ts [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+  onChange = (value: any) => { };
+  onTouched = () => { };
+
+
+  constructor(public fb: FormBuilder) { }
+
+  ngOnInit(): void {
+    this.colorsForm.valueChanges.subscribe((value) => {
```
Comment (vsvanidze) [inline src/app/components/colors/colors.component.ts]: we need to unsubscribe when this component will destroy

use take until strategy https://www.digitalocean.com/community/tutorials/angular-takeuntil-rxjs-unsubscribe

**Review by vsvanidze — APPROVED** [review]

**Pull request merged**